src/extension/polling.ts

```
@@ -103,16 +106,57 @@ export async function refreshWorkflowState(ctx: PollingContext, webview: vscode.
+ * is unchanged since the last post (nothing to show). Cuts the idle cost of the
+ * 30s loop — relevant because `.gsd` is often Dropbox-synced.
+ */
+async function pollWorkflowStateGated(ctx: PollingContext, webview: vscode.Webview, sessionId: string): Promise<void> {
+  // No point re-posting to a hidden webview — the badge isn't visible and the
+  // next resolve/rebind primes state from scratch.
+  if (!ctx.isWebviewVisible(sessionId)) return;
+
+  const cwd = vscode.workspace.workspaceFolders?.[0]?.uri.fsPath || process.cwd();
+  const statePath = path.join(cwd, ".gsd", "STATE.md");
+
+  let mtimeMs = 0;
```

ESLint

[error] The value assigned to 'mtimeMs' is not used in subsequent statements.

(no-useless-assignment)